Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -280,36 +280,63 @@ public extension Tensor {
     @inlinable
     @differentiable(
         wrt: self,
-        vjp: _vjpTransposed(withPermutations:) where Scalar: TensorFlowFloatingPoint)
+        vjp: _vjpTransposed(permutation:) where Scalar: TensorFlowFloatingPoint)
+    @available(*, deprecated, message: "Please use transposed(permutation:) instead.")
     func transposed(withPermutations permutations: Tensor<Int32>) -> Tensor {
-        return Raw.transpose(self, perm: permutations)
+        return transposed(permutation: permutations)
+    }
+
+    @inlinable
+    @differentiable(
+            wrt: self,
+            vjp: _vjpTransposed(permutation:) where Scalar: TensorFlowFloatingPoint)
+    func transposed(permutation perm: Tensor<Int32>) -> Tensor {
+        return Raw.transpose(self, perm: perm)
     }
 
     /// Returns a transposed tensor, with dimensions permuted in the specified order.
     @inlinable
     @differentiable(
-        wrt: self,
-        vjp: _vjpTransposed(withPermutations:) where Scalar: TensorFlowFloatingPoint)
+            wrt: self,
+            vjp: _vjpTransposed(permutation:) where Scalar: TensorFlowFloatingPoint)
+    @available(*, deprecated, message: "Please use transposed(permutation:) instead.")
     func transposed(withPermutations permutations: [Int]) -> Tensor {
-        let permutations = permutations.map(Int32.init)
-        return transposed(withPermutations: Tensor<Int32>(permutations))
+        return transposed(permutation: permutations)
+    }
+
+    /// Returns a transposed tensor, with dimensions permuted in the specified order.
+    @inlinable
+    @differentiable(
+        wrt: self,
+        vjp: _vjpTransposed(permutation:) where Scalar: TensorFlowFloatingPoint)
+    func transposed(permutation perm: [Int]) -> Tensor {
```

Comment:
```suggestion
    func transposed(permutation: [Int]) -> Tensor {
```

Reply:
You are right, it is redundant this one too.